File: sty_algorithm/src/main/scala/com/sty/algorithm/leetcode/editor/cn/[349]两个数组的交集.scala
//leetcode submit region begin(Prohibit modification and deletion)
import scala.collection.mutable
object Solution {
  def intersection(nums1: Array[Int], nums2: Array[Int]): Array[Int] = {
    val map: mutable.Map[Int, Int] = mutable.HashMap()

    for (elem <- nums1) {
      map.put(elem, 0)
    }

    var res: mutable.HashSet[Int] = mutable.HashSet()
    for (elem <- nums2) {
      if (map.contains(elem)) {
        res = res + elem
      }
    }

    res.toArray
  }
}
//leetcode submit region end(Prohibit modification and deletion)

File: sty_algorithm/src/main/java/com/sty/algorithm/leetcode/editor/cn/[349]两个数组的交集.java
package com.sty.algorithm.leetcode.editor.cn;

import java.util.HashSet;
import java.util.Set;

//leetcode submit region begin(Prohibit modification and deletion)
class Solution349 {
    public int[] intersection(int[] nums1, int[] nums2) {

        Set<Integer> set = new HashSet<>(nums1.length);
        for (int i : nums1) {
            set.add(i);
        }

        Set<Integer> res = new HashSet<>(nums1.length);
        for (int i : nums2) {
            if (set.contains(i)) {
                res.add(i);
            }
        }

        return res.stream().mapToInt(Integer::intValue).toArray();
    }
}
//leetcode submit region end(Prohibit modification and deletion)
